sia/host/contracts.go

```diff
@@ -0,0 +1,214 @@
+package host
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"os"
+	"strconv"
+
+	"github.com/NebulousLabs/Sia/consensus"
+	"github.com/NebulousLabs/Sia/encoding"
+	"github.com/NebulousLabs/Sia/hash"
+	"github.com/NebulousLabs/Sia/sia/components"
+)
+
+// ContractEntry houses a single contract with its id - you cannot derive the
+// id of a contract without having the transaction. Rather than keep the whole
+// transaction, we store only the id.
+type ContractEntry struct {
+	ID       consensus.ContractID
+	Contract consensus.FileContract
+}
+
+func (h *Host) getFilename() string {
+	h.fileCounter++
+	return strconv.Itoa(h.fileCounter)
+}
```

Comment:
To me, a function named `get` implies that no state is being changed. Maybe it should be `nextFilename`?